### PR TITLE
 graph pass

### PR DESCRIPTION
**Pull request by Mark-Partola**

### CODE REVIEW

File: graph-key/src/app.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  svgCanvas.addEventListener("mousemove", moveEdge);
+  svgCanvas.addEventListener("mouseup", cancelMove);
+  svgCanvas.addEventListener("touchmove", moveEdge);
+  svgCanvas.addEventListener("touchend", cancelMove);
+}
+
+svgCanvas.addEventListener("touchstart", startMove);
+svgCanvas.addEventListener("mousedown", startMove);
```
Comment (sandrina-p) [inline graph-key/src/app.js]: That's the way to go, have both touch and mouse events! ✨

File: graph-key/src/app.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import "../style.css";
+import "../graph-key.css";
+
+const config = {
+  el: document.querySelector(".graph-key"),
+  pointsCount: 4,
+  radius: 10,
+  size: 300,
+  padding: 30
+};
```
Comment (sandrina-p) [inline graph-key/src/app.js]: I really like what I see here, didn't see anything wrong so far! What are your plans for this?

What's missing? :)

File: graph-key/src/utils.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    x: element.getAttribute("cx"),
+    y: element.getAttribute("cy")
+  };
+}
+
+export function generateSvgPathAttribute(points) {
+  return points.reduce(
+    (acc, curr) => acc + `${!acc ? "M" : "L"}${curr.x} ${curr.y}`,
+    ""
+  );
```
Comment (sandrina-p) [inline graph-key/src/utils.js]: The way you deal with svg creation and modification is clean, well done!

File: graph-key/graph-key.css [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  }
+}
+
+@keyframes shake {
+  0% {
+    transform: translate(1px, 1px);
+  }
+
+  100% {
+    transform: translate(-1px, -1px);
```
Comment (sandrina-p) [inline graph-key/graph-key.css]: I barely noticed this animation, maybe you could increase it by 2-3 px :)

File: graph-key/src/app.js (hunk 2 above)
Comment (Mark-Partola) [reply graph-key/src/app.js]: I don't have any plans, just wanted to make this))

What do you mean about missing?)

File: graph-key/graph-key.css (hunk 4 above)
Comment (Mark-Partola) [reply graph-key/graph-key.css]: Yes, I think it would be better :)

File: graph-key/src/app.js (hunk 2 above)
Comment (sandrina-p) [reply graph-key/src/app.js]: Perhaps when you do the first combination it saves it and then only one second combination it turns green or red if it matches the first. And a reset button to start over

**Pull request merged**